src/Autocomplete/src/Controller/EntityAutocompleteController.php

```diff
@@ -54,4 +66,48 @@ public function __invoke(string $alias, Request $request): Response
             'next_page' => $nextPage,
         ]);
     }
+
+    /**
+     * @return array<string, scalar|array|null>
+     */
+    private function getExtraOptions(Request $request): array
+    {
+        if (!$request->query->has(self::EXTRA_OPTIONS)) {
+            return [];
+        }
+
+        $extraOptions = $this->getDecodedExtraOptions($request->query->get(self::EXTRA_OPTIONS));
+
+        if (!\array_key_exists(AutocompleteChoiceTypeExtension::CHECKSUM_KEY, $extraOptions)) {
+            throw new BadRequestHttpException('The extra options are missing the checksum.');
+        }
+
+        $this->validateChecksum($extraOptions[AutocompleteChoiceTypeExtension::CHECKSUM_KEY], $extraOptions);
+
+        return $extraOptions;
+    }
+
+    /**
+     * @return array<string, scalar>
+     */
+    private function getDecodedExtraOptions(string $extraOptions): array
+    {
+        return json_decode(base64_decode($extraOptions), true);
```

Comment:
Sorry for the late comment, but i think we should: 

* add  `JSON_THROW_ON_ERROR` flag here to throw if anything goes wrong (or we will have a type error, as array_key_exists will throw a TypeError if null is passed as second argument)
* probably replace `$request->query->get(self::EXTRA_OPTIONS)` with `$request->query->getString(..)` to avoid the same problem